tests/D2L.CodeStyle.Analyzers.Test/Specs/ImmutabilityAnalyzer.cs

```diff
@@ -363,4 +363,79 @@ public class MutableClassImmutableState<[Objects.Immutable] T> {
 			private readonly T m_field;
 		}
 	}
+
+	class ConcreteGenericTests {

```

Comment:
So I was expecting to see some tests like:

```
		[Objects.Immutable]
		interface ImmutableInterface<[Objects.Immutable] T> {
		}

                class SomeClass {
                    static void Method() {
                          ImmutableInterface<MutableClass> blah = default;
                    } 
                }
```

Which shouldn't compile.  We need to every ```GenericNameSyntax```

Reply:
![image](https://user-images.githubusercontent.com/12101120/42583370-d129ed18-84fe-11e8-88a9-880a9a40835d.png)

Reply:
![image](https://user-images.githubusercontent.com/12101120/42583416-e4f3c2b0-84fe-11e8-8d7f-c031bbc06d19.png)

Reply:
It hasn't historically been the case that the analyzer concerns itself with things used in a mutable context.  So, if something said Foo<[Immutable]T>, but passed a mutable type to it, as long as Foo wasn't being held in an immutable context, we couldn't actually stop it from happening.

In order to be able to catch that, we'll have to open up the scanner to whole of additional scan entry points.  I'll see what i can do.

Reply:
Given that being able to detect mutable classes passed to immutable type constraints in a mutable context is new functionality, does this commit suffice and I can start a new PR that builds on this to add that functionality?